src/audio/kpb.c

```diff
@@ -521,7 +521,7 @@ static int kpb_prepare(struct comp_dev *dev)
 	}
 
 	if (!kpb->sel_sink || !kpb->host_sink) {
-		comp_info(dev, "kpb_prepare(): could not find sinks: sel_sink %d host_sink %d",
+		comp_info(dev, "kpb_prepare(): could not find sinks: sel_sink %lu host_sink %lu",
 			  (uintptr_t)kpb->sel_sink, (uintptr_t)kpb->host_sink);
```

Comment:
Paranoia: I don't know how sophisticated the SOF native printf library is and whether it supports "%lu".  Since the arguments are already being typecast, it might be more robust to just cast them to "int", which is known to work with "%d" in all cases.

Pedantry: the original format string was using "%d", so if the arguments had values >= 0x80000000 they would have been emitted as negative numbers.  That's probably an upstream bug, since the arguments are unsigned, but technically this does change behavior.

Reply:
those are already being cast to unitptr_t, not sure how this works in SOF with %d. Something is not right.